### PR TITLE
Refactor metadata Table tests to use a common parent

### PR DESCRIPTION
**Pull request by golammott**
Attempting to address #659. Refactored `TestIcebergSourceHiveTables` and `TestIcebergSourceHadoopTables` to use common parent`TestIcebergSourceTablesBase`. 

All test logic stayed the same but combined these to remove redundant code where possible. Let me know any thoughts!

### CODE REVIEW

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceHadoopTables.java [hunk 1]
```diff
@@ -101,31 +79,7 @@ public void testEntriesTable() throws Exception {
...
-    InputFile manifest = table.io().newInputFile(table.currentSnapshot().manifests().get(0).path());
-    List<GenericData.Record> expected;
-    try (CloseableIterable<GenericData.Record> rows = Avro.read(manifest).project(entriesTable.schema()).build()) {
-      expected = Lists.newArrayList(rows);
-    }
-
-    Assert.assertEquals("Entries table should have one row", 1, expected.size());
-    Assert.assertEquals("Actual results should have one row", 1, actual.size());
-    TestHelpers.assertEqualsSafe(entriesTable.schema().asStruct(), expected.get(0), actual.get(0));
+    super.testEntriesTable(table, entriesTable, tableLocation, tableLocation + "#entries");
```
Comment (rdblue) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceHadoopTables.java]: It isn't necessary to use `super` here. Could you remove those?

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceHiveTables.java [hunk 2]
```diff
@@ -695,63 +298,13 @@ public synchronized void testHiveAllManifestsTable() throws Exception {
     TableIdentifier tableIdentifier = TableIdentifier.of("db", "manifests_test");
     try {
       Table table = catalog.createTable(
-          tableIdentifier,
-          SCHEMA,
-          PartitionSpec.builderFor(SCHEMA).identity("id").build());
+              tableIdentifier,
```
Comment (rdblue) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceHiveTables.java]: Can you revert these changes? The indentation was correct.

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceHadoopTables.java [hunk 3]
```diff
@@ -185,92 +97,23 @@ public void testFilesTable() throws Exception {
...
-    Table filesTable = TABLES.load(tableLocation + "#files");
+    try {
+      Table table = TABLES.create(SCHEMA, PartitionSpec.builderFor(SCHEMA).identity("id").build(), tableLocation);
 
-    List<SimpleRecord> records = Lists.newArrayList(
-        new SimpleRecord(1, "a"),
-        new SimpleRecord(2, "b")
-     );
+      table.updateProperties()
+              .set(TableProperties.SNAPSHOT_ID_INHERITANCE_ENABLED, "true")
```
Comment (rdblue) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceHadoopTables.java]: Continuing indentation is 4 spaces, not 8. Could you update this?

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceHiveTables.java [hunk 4]
```diff
@@ -147,32 +131,7 @@ public synchronized void testHiveEntriesTable() throws Exception {
...
-      List<GenericData.Record> expected;
-      try (CloseableIterable<GenericData.Record> rows = Avro.read(manifest).project(entriesTable.schema()).build()) {
-        expected = Lists.newArrayList(rows);
-      }
-
-      Assert.assertEquals("Entries table should have one row", 1, expected.size());
-      Assert.assertEquals("Actual results should have one row", 1, actual.size());
-      TestHelpers.assertEqualsSafe(entriesTable.schema().asStruct(), expected.get(0), actual.get(0));
-
+      super.testEntriesTable(table, entriesTable, tableIdentifier.toString(), "db.entries_test.entries");
```
Comment (rdblue) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceHiveTables.java]: The Hadoop table tests use before/after methods to create and clean up the table. Could this use an after method that calls `DROP TABLE IF EXISTS` to avoid the try/finally in each method?

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java [hunk 5]
```diff
@@ -0,0 +1,586 @@
...
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.TableIdentifier;
+import org.junit.Assert;
+
+public abstract class TestIcebergSourceTablesBase {
+
+  protected static SparkSession spark;
+
+  public void testEntriesTable(Table table, Table entriesTable,
```
Comment (rdblue) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java]: The problem with passing in the table is that this creates a dependency between the caller and this code: this assumes that the caller has created the table with the correct schema, for example.

Instead, what about creating abstract methods to create the table, drop the table if it exists, and load the entries table? That way, you could write these test cases as `@Test` methods that are inherited and the Hive and Hadoop subclasses would just need to implement a few methods.

**Comment by rdblue** [comment]
Thanks for working on this, @golammott! It's a great start, but I think we probably want to refactor a bit more; see my last comment.

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java (hunk 5 above)
Comment (golammott) [reply spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java]: Thanks for the feedback @rdblue! Just updated the PR to use the abstract methods you described, hopefully this is closer to what you are looking for.

**Comment by rdblue** [comment]
Looks great, @golammott! I merged this.
